### ISSUE TITLE
WineASIO not an option despite patching

### ISSUE THREAD

**Issue by theNizo** [theNizo/linux_rocksmith#28]
* Which distro do you use? - Arch
* Which Proton or wine version do you use? - Tried both Proton 7 and Proton 8
* Do you use pipewire? - Yes
* What appears to be the problem? (Describe it as best as you can)
  * (deleted prefix inbetween testing)
  * normal wine, pfx = `~/wine-test`, patch it - VBASIOTester shows WineASIO.
  * Proton, pfx = `~/wine-test`, patch it - VBASIOTester shows WineASIO.
  * Generate 221680 Prefix via starting Rocksmith, patch it, then use normal wine - VBASIOTester CAN NOT find WineASIO
  * Generate 221680 Prefix via starting Rocksmith, patch it, then use Proton - VBASIOTester CAN NOT find WineASIO
* Did you notice any other unexpected behavior? - Not unexpected, but Steam Linux Runtime got updated from Soldier to Sniper. I'm not sure if this has an impact though.
* Did you do any of the steps differently or leave them out? - I doubt it, since I've read it often enough

So, basically, the prefixes generated by Steam are broken.

This is new behavior.

Does anyone else experience this? If you test that, please backup your working prefix first.

**Comment by cyrilpicard** [comment]
I can't reproduce on my system. WineASIO is found by VBASIOTester for both prefixes generated by Wine and Steam/Proton Experimental.

Can you describe how you patch the prefix ?

**Comment by theNizo** [comment]
Okay, might be an issue on my side then. I'll try it on my laptop too.

> Can you describe how you patch the prefix ?

1. Delete or rename `$STEAMLIBRARY/compatdata/221680`
2. Start Rocksmith from Steam with the button that says "Play", first time setup is performed, wait until the game's window appears
3. Run `WINEPREFIX=$STEAMLIBRARY/steamapps/compatdata/221680/pfx regsvr32 /usr/lib32/wine/i386-windows/wineasio.dll`
4. registers successfully, according to terminal output.
5. Run VBASIOTester via `WINEPREFIX=$STEAMLIBRARY/steamapps/compatdata/221680/pfx $PROTON/bin/wine  ~/Downloads/VBASIOTest/VBASIOTest32.exe`
6. Notice no difference.

Done both with Proton 7.0 and Proton 8.0

Things I could think about trying (honestly, I have no idea here):

* Protontricks, although it should just be a winetricks wrapper AFAIK
* Proton-GE

**Comment by theNizo** [comment]
Replicated it on my laptop as well, both Arch

**Comment by cyrilpicard** [comment]
I can reproduce the issue doing the steps above.

Here are the steps I use to configure asio in wine prefix (which I used with success in my previous test) :
- copy wineasio.dll in $PROTON/lib/wine/i386-windows
- copy wineasio.dll.so in $PROTON/lib/wine/i386-unix
- register wineasio using proton's wine (in this case the dll is registered but not copied to the prefix, we will copy it manually in the next step)
`WINEPREFIX=$STEAMLIBRARY/compatdata/221680/pfx $PROTON/bin/wine regsvr32 $PROTON/lib/wine/i386-windows/wineasio.dll`
- copy wineasio.dll to wine prefix
`cp $PROTON/lib/wine/i386-windows/wineasio.dll $STEAMLIBRARY/compatdata/221680/pfx/drive_c/windows/system32/`
`cp $PROTON/lib/wine/i386-windows/wineasio.dll $STEAMLIBRARY/compatdata/221680/pfx/drive_c/windows/syswow64/`
- Run VBASIOTester
`WINEPREFIX=$STEAMLIBRARY/compatdata/221680/pfx $PROTON/bin/wine VBASIOTest32.exe`
- Wineasio is found and works

I prefer this method since it does not rely on standalone wine (it justs use the one provided by Proton) and it does not require wineasio to be installed systemwide (it is just copied in Proton lib directory). I think prefixes generated by Steam are not meant to be run outside Steam.

**Comment by theNizo** [comment]
It crashes at start. That's a step forward! I'll keep trying.

**Comment by theNizo** [comment]
That works. I'll adjust the guide. Thank you very much @cyrilpicard !
